### ISSUE TITLE
Modules that are created by using Functors are not indexed

### ISSUE THREAD

**Issue by Coobaha** [reasonml-editor/reasonml-idea-plugin#205]
### Description

```

module type Comparable = {
  type t;
  let equal: (t, t) => bool;
};

module MakeSet = (Item: Comparable) => {
  /* let's use a list as our naive backing data structure */
  type backingType = list(Item.t);
  let empty = [];
  let add = (currentSet: backingType, newItem: Item.t) : backingType =>
    /* if item exists */
    if (List.exists((x) => Item.equal(x, newItem), currentSet)) {
      currentSet /* return the same (immutable) set (a list really) */
    } else {
      [
        newItem,
        ...currentSet /* prepend to the set and return it */
      ]
    };
};

module IntPair = {
  type t = (int, int);
  let equal = ((x1, y1), (x2, y2)) => x1 == x2 && y1 == y2;
  let create = (x, y) => (x, y);
};

/* IntPair abides by the Comparable signature required by MakeSet */
module SetOfIntPairs = MakeSet(IntPair);

SetOfIntPairs.<caret>
```
results in no suggestions.

It is possible that return type is also defined:

```
module type Comparable = ...

module type MakeSetType = (Item: Comparable) => {
  type backingType;
  let empty: backingType;
  let add: (backingType, Item.t) => backingType;
};

module MakeSet: MakeSetType = (Item: Comparable) => {
  ...
};
```

**Expected behavior:**

If no interface is provided (.rei or return type) - index whole returned module
If interface is found - it should be used instead

**Comment by donut** [comment]
This problem makes [bs-webapi](https://github.com/reasonml-community/bs-webapi-incubator) very difficult to use with this plugin as basically all of its libraries are generated with functors. If this problem was solved, I would be able to use it. Until then I'm stuck with VSCode and [reason-language-server](https://github.com/jaredly/reason-language-server).

**Comment by giraud** [comment]
this will take some time, but next task on the list

**Comment by giraud** [comment]
![image](https://user-images.githubusercontent.com/239612/84153337-08d84a00-aa66-11ea-941e-3fa75aa317b3.png)

**Comment by donut** [comment]
Yay! I'm so excited, this missing feature was the only thing keeping me on VSCode for OCaml/Reason. Thank you!

**Comment by giraud** [comment]
yep, it seems to work with webapi:
![image](https://user-images.githubusercontent.com/239612/84252316-bacb5100-ab0e-11ea-8985-0e4b1141ff85.png)

**Comment by giraud** [comment]
Fixed in 0.90
